### PR TITLE
Update Finagle to 6.36

### PR DESCRIPTION
**Pull request by vkostyukov**
This PR updates Finagle to 6.36 and also makes some misc changes:

1. Make `DateTimeFormat` a thread-local since it's not thread-safe and it's not fair to share it
2. Use one-jar plugin to run the program instead of reusing the sbt's JVM instance

Please let me know if there anything I can do/change to make this PR make it to the Round 13.

### CODE REVIEW

File: frameworks/Scala/finagle/src/main/scala/Main.scala [hunk 1]
```diff
@@ -13,14 +13,17 @@ import com.twitter.io.Buf
 object Main extends App {
 
   val mapper: ObjectMapper = new ObjectMapper().registerModule(DefaultScalaModule)
-  val dateFormat: DateFormat = new SimpleDateFormat("E, dd MMM yyyy HH:mm:ss z")
+
+  val dateFormat: ThreadLocal[DateFormat] = new ThreadLocal[DateFormat] {
+    override def initialValue: DateFormat = new SimpleDateFormat("E, dd MMM yyyy HH:mm:ss z")
```
Comment (cacoco) [inline frameworks/Scala/finagle/src/main/scala/Main.scala]: The format here should be "E, dd MMM yyyy HH:mm:ss 'GMT'". See the issue we hit and the pull-request here: https://github.com/twitter/finatra/pull/312

Comment (vkostyukov) [inline frameworks/Scala/finagle/src/main/scala/Main.scala]: Thanks @cacoco! I updated the PR to use request API `req.date = ???` instead that uses the right date format.

**Comment by knewmanTE** [comment]
@vkostyukov looks good! we'll do our best to get this into round 13!

**Comment by vkostyukov** [comment]
Thank you, @knewmanTE!

**Pull request merged**